src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -693,8 +693,14 @@ void Compiler::eeSetLVdone()
         eeDispVars(info.compMethodHnd, eeVarsCount, (ICorDebugInfo::NativeVarInfo*)eeVars);
     }
 #endif // DEBUG
-
-    info.compCompHnd->setVars(info.compMethodHnd, eeVarsCount, (ICorDebugInfo::NativeVarInfo*)eeVars);
+    if (0 < eeVarsCount)
+    {
+        info.compCompHnd->setVars(info.compMethodHnd, eeVarsCount, (ICorDebugInfo::NativeVarInfo*)eeVars);
```

Comment:
This is changing the contract of this JIT-EE function, before this we were calling `setVars(info.compMethodHnd, 0, nullptr)` in the case when there are no variable live ranges, through this path:

https://github.com/dotnet/runtime/blob/689350960f9cbb76c7aa469f129b5e7227324dae/src/coreclr/jit/codegencommon.cpp#L6813-L6819